utility/src/main/java/org/oppia/util/parser/SvgDecoder.kt

```diff
@@ -0,0 +1,32 @@
+package org.oppia.util.parser
+
+import com.bumptech.glide.load.Options
+import com.bumptech.glide.load.ResourceDecoder
+import com.bumptech.glide.load.engine.Resource
+import com.bumptech.glide.load.resource.SimpleResource
+import com.caverock.androidsvg.SVG
+import com.caverock.androidsvg.SVGParseException
+import java.io.IOException
+import java.io.InputStream
+
+/** Decodes an SVG internal representation from an {@link InputStream}. */
+class SvgDecoder : ResourceDecoder<InputStream?, SVG?> {
+
+  override fun handles(source: InputStream, options: Options): Boolean {
+    return true
+  }
+
+  override fun decode(
+    source: InputStream,
+    width: Int,
+    height: Int,
+    options: Options
+  ): Resource<SVG?>? {
+    return try {
+      val svg = SVG.getFromInputStream(source)
```

Comment:
Does/should this close the input stream? Glide's documentation hopefully indicates whether we're responsible for closing it.

Reply:
Closed input stream.